src/openms/include/OpenMS/FILTERING/ID/IDFilter.h

```diff
@@ -542,6 +542,15 @@ namespace OpenMS
                   items.end());
     }
 
+    /// Move items that satisfy a condition to a container (e.g. vector)
+    template <class Container, class Predicate>
+    static void moveMatchingItems(Container& items, const Predicate& pred, Container& target)
```

Comment:
maybe consider a move_if  implementation as described in https://stackoverflow.com/questions/32155810/move-all-elements-which-satisfy-some-condition-from-one-container-to-another-i

Reply:
but not that important

Reply:
I actually implemented the first answer of the link. Since we have to shrink the vector, I used the partition approach.